source/concurrent/RecycleBin.ooc

```diff
@@ -10,30 +10,40 @@ use concurrent
 
 RecycleBin: class <T> {
 	_free: Func (T)
-	_list: SynchronizedList<T>
+	_list: VectorList<T>
 	_size: Int
 	count ::= this _list count
 	isFull ::= this count == this _size
 	isEmpty ::= this _list empty
-	init: func (=_size, =_free) { this _list = SynchronizedList<T> new(_size) }
+	_mutex := Mutex new()
+	init: func (=_size, =_free) { this _list = VectorList<T> new(_size) }
 	free: override func {
 		this clear()
 		this _list free()
 		(this _free as Closure) free(Owner Receiver)
+		this _mutex free()
 		super()
 	}
 	add: func (object: T) {
+		this _mutex lock()
 		if (this isFull)
 			this _free(this _list remove(0))
 		this _list add(object)
+		this _mutex unlock()
 	}
 	search: func (matches: Func (T) -> Bool) -> T {
-		index := this _list search(matches)
 		result: T = null
-		(index > -1) ? this _list remove(index) : result
+		this _mutex lock()
```

Comment:
It was here https://github.com/magic-lang/ooc-kean/pull/1821#issuecomment-229643939 
Yes, it does matter for generics, because this line `result: T = null` can trigger the "constructor" of  the `Pointer_Class` due to lazy initialization. This constructor can then modify a static bool flag.